tests/run-make/optimization-remarks-dir/rmake.rs

```diff
@@ -0,0 +1,40 @@
+// In this test, the function `bar` has #[inline(never)] and the function `foo`
+// does not. This test outputs LLVM optimization remarks twice - first for all
+// functions (including `bar`, and the `inline` mention), and then for only `foo`
+// (should not have the `inline` mention).
+// See https://github.com/rust-lang/rust/pull/113040
+
+use run_make_support::{
+    has_extension, has_prefix, invalid_utf8_contains, invalid_utf8_not_contains, rustc,
+    shallow_find_files,
+};
+
+fn main() {
+    rustc()
+        .opt()
+        .input("foo.rs")
+        .crate_type("lib")
+        .arg("-Cremark=all")
+        .arg("-Zremark-dir=profiles_all")
+        .run();
+
+    for file in shallow_find_files("profiles_all", |path| {
+        has_prefix(path, "foo") && has_extension(path, "yaml")
```

Comment:
Suggestion: can you either pull out a temp variable for `shallow_find_files` here and below or add a comment on what the files found are? I assume they are remark files?

Reply:
I did, and I also added the `not_contains` filter for consistency instead of the if statement checking if `codegen` is not in the filename.

@rustbot review